docs/architecture/porting-existing-aspnet-apps/example-migration-eshop.md

```diff
@@ -0,0 +1,846 @@
+
+![Figure 4-12](media/Figure4-12.png)
+
+**Figure 4-12. Ported *eShop* app running locally with mock data.**
+
+#### Migrate app settings
+
+ASP.NET Core uses a new [configuration system](https://docs.microsoft.com/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.2&preserve-view=true), which by default leverages an *appsettings.json* file. By using `CreateDefaultBuilder` in *Program.cs*, the default configuration is already set up in the app. To access configuration, classes just need to request it in their constructor. The `Startup` class is no exception. To start accessing configuration in `Startup` and the rest of the app, request an instance of `IConfiguration` from its constructor:
+
+```csharp
+public Startup(IConfiguration configuration)
+{
+    Configuration = configuration;
+}
+
+public IConfiguration Configuration { get; }
+```
+
+The original app referenced its settings using `ConfigurationManager.AppSettings`. A quick search for all references of this term yields the set of settings the new app needs. There are only two:
+
+- `UseMockData`
+- `UseCustomizationData`
+
+If your app has more complex configuration, especially if it's using custom configuration sections, you'll probably want to create and bind objects to different parts of your app's configuration. These types can then be accessed using the [Options pattern](https://docs.microsoft.com/aspnet/core/fundamentals/configuration/options?view=aspnetcore-2.2&preserve-view=true). However, as noted in the referenced doc, this pattern shouldn't be used in `ConfigureServices`. Instead the ported app will reference the `UseMockData` configuration value directly.
+
+First, modify the ported app's `appsettings.json` file and add the two settings in the root:
+
+```json
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Warning"
+    }
+  },
+  "AllowedHosts": "*",
+  "UseMockData": "true",
+  "UseCustomizationData" :  "true"
+}
+```
+
+Now, modify `ConfigureServices` to access the `UseMockData` setting from the `Configuration` property (where previously we set the value to `true`):
+
+```csharp
+  bool useMockData = Configuration.GetValue<bool>("UseMockData");
+```
+
+At this point, the setting is pulled from configuration. The other setting, `UseCustomizationData`, is used by the `CatalogDBInitializer` class. When you first ported this class, you commented out the access to `ConfigurationManager.AppSettings["UseCustomizationData"]`. Now it's time to modify it to use ASP.NET Core configuration. Modify the constructor of `CatalogDBInitializer` as follows:
+
+```csharp
+  // add using Microsoft.Extensions.Configuration
+  public CatalogDBInitializer(CatalogItemHiLoGenerator indexGenerator,
+      IConfiguration configuration)
+  {
+      this.indexGenerator = indexGenerator;
```

Comment:
I'd really like to clean this up, private fields should be prefixed with `_` to avoid using `this` as a qualifier.

Reply:
For now it's literally the code from the repo which uses the `this.` convention. I agree with you and prefer `_` but I don't necessarily want to have to reformat the whole sample project. And I think ideally we want these code listings to accurately reflect the referenced sample. What do you suggest?

Reply:
I thought I saw the correct naming convention in other examples in this eBook, no? I could be mistaken. I'd really like to get this fixed, let's make it a post-publish effort.